src/lib/NodeSmartLinkProvider.ts

```diff
@@ -234,13 +237,19 @@ export class NodeSmartLinkProvider {
     if (attributes.has(DataAttribute.ElementCodename)) {
       event.preventDefault();
 
-      this.events.emit(NodeSmartLinkProviderEventType.ElementClicked, {
+      const data: Partial<IElementClickedMessageData> = {
         projectId: attributes.get(DataAttribute.ProjectId),
         languageCodename: attributes.get(DataAttribute.LanguageCodename),
         itemId: attributes.get(DataAttribute.ItemId),
         contentComponentId: attributes.get(DataAttribute.ComponentId),
         elementCodename: attributes.get(DataAttribute.ElementCodename),
-      });
+      };
+
+      const metadata: IElementClickedMessageMetadata = {
+        targetRect: (event.target as HTMLElement).getBoundingClientRect(),
```

Comment:
What if `event.target` is not HTMLElement? This could use some type check or assert with a better message than a null reference that would fire otherwise

Reply:
I added assert there.